packages/core/src/Serialization/PlutusData/PlutusData.ts

```diff
@@ -370,7 +370,43 @@ export class PlutusData {
    * @returns The Uint8Array or undefined if it can not be 'down cast'.
    */
   asBoundedBytes(): Uint8Array | undefined {
-    return this._bytes;
+    return this.#bytes;
+  }
+
+  /**
+   * Indicates whether some other PlutusData is "equal to" this one.
+   *
+   * @param other The other object to be compared.
+   * @returns true if objects are equals; otherwise false.
+   */
+  // eslint-disable-next-line complexity
+  equals(other: PlutusData): boolean {
+    let result = false;
+    switch (this.#kind) {
+      case PlutusDataKind.Bytes:
+        return HexBlob.fromBytes(this.#bytes!) === HexBlob.fromBytes(other.#bytes!);
```

Comment:
Possible optimization here (nitpick :sweat_smile:)
```suggestion
        return this.#bytes && this.#bytes.length === other.#bytes?.length && this.#bytes.every((value, index) => value === other.#bytes![index]);
```